service_crategen/src/main.rs

```diff
@@ -0,0 +1,287 @@
+    }
+
+    let service_versions: Vec<(String, String)> = {
+        if let Some(services) = matches.values_of("services_list") {
+            services.map(|s| {
+                let mut service_parts = s.splitn(2, "@");
+                let name = service_parts.next().expect(&format!("Invalid service value {}. Must be in format name@version.", s)).to_owned();
+                let version = service_parts.next().expect(&format!("Invalid service value {}. Must be in format name@version.", s)).to_owned();
+                (name, version)
+            }).collect()
+        } else if let Some(services_config) = matches.value_of("services_config") {
+            let contents = File::open(services_config).and_then(|mut f| {
+                let mut contents = String::new();
+                f.read_to_string(&mut contents).map(|_| contents)
+            }).expect("Unable to read services configuration file.");
+
+            let parsed: BTreeMap<String, ServiceConfig> = serde_json::from_str(&contents).expect("Unable to parse services configuration file.");
+            parsed.iter().map(|(k, v)| (k.clone(), v.version.clone())).collect()
+        } else {
+            unreachable!("Arg parsing failed. Did not have services config or services list.");
+        }
+    };
+
+    service_versions.par_iter().for_each(|&(ref name, ref service_version)| {
+        let service = Service::load(name, service_version);
+
+        if let Err(ref e) = service {
+            println!("Failed to load service {}: {}", name, e);
+            return;
+        }
+        
+        let service = service.as_ref().unwrap();
+
+        let crate_dir = out_dir.join(&name);
+        let crate_name = format!("rusoto_{}", &name);
+
+        println!("Generating crate for {} @ {}...", service.metadata.service_full_name, service.metadata.api_version);
+
+        if !crate_dir.exists() {
+            fs::create_dir(&crate_dir).expect(&format!("Unable to create directory at {}", crate_dir.display()));
+        }
+
+        let service_dependencies = get_dependencies(&service, &version);
+
+        let extern_crates = service_dependencies.iter().map(|(k, _)| {
+            if k == "xml-rs" {
+                return "extern crate xml;".into();
+            }
+            let safe_name = k.replace("-", "_");
+            let use_macro = k == "serde_derive" || k == "log" || k == "lazy_static";
+            if use_macro {
+                return format!("#[macro_use]\nextern crate {};", safe_name);
+            }
+            format!("extern crate {};", safe_name)
+        }).collect::<Vec<String>>().join("\n");
+
+        let mut cargo_manifest = OpenOptions::new()
+            .write(true)
+            .truncate(true)
+            .create(true)
+            .open(crate_dir.join("Cargo.toml"))
+            .expect("Unable to write Cargo.toml");
+        
+        let manifest = cargo::Manifest {
+            package: cargo::Metadata {
```

Comment:
As with the above comment, maybe this preamble should all be in a `.toml` file somewhere that gets read in and either just straight regurgitated back out, or parsed and manipulated if needs be.  It'd make tweaking it a bit easier.